src/openforms/utils/tests/cache.py

```diff
@@ -2,5 +2,10 @@
 
 
 def clear_caches():
-    for cache in caches.all():
+    for alias in caches:
+        # clearing the distributed session locks causes issues when tests are run
+        # in parallel (--parallel X)
+        if alias == "portalocker":
+            continue
+        cache = caches[alias]
```

Comment:
Is there a way we could only clear the Redis cache values related to the current test? Not clearing _all_ cache values from Redis makes sense, but perhaps we could clear only the relevant keys? Not sure if this is possible.

My idea was to "record" all keys set during a test, and use `delete_many` on these keys as a cleanup. Probably too hard to implement and not worth the effort.

Reply:
no, it's near impossible to trace down which keys are all used and I really don't want this implementation detail to leak into the test setup since every time a library adds caching we need to be aware of the keys used. We also often need access to local-scoped variables for dynamic keys that we simply don't have.

The premise for tests is still that they should be fully isolated, meaning that nuking redis/cache entirely after a test-run is the semantically correct approach.

> My idea was to "record" all keys set during a test, and use delete_many on these keys as a cleanup. Probably too hard to implement and not worth the effort.

I don't think this is feasible. Databases are much nicer in that regard in the sense that a transaction can just be rolled back.

Personally I was thinking in getting the test process ID/index and using that to point to a particular redis DB so that each test process has its own redis database, but settings are loaded before the multiprocessing pool is started and tests are distributed to it. Django actually sort-of monkeypatches the DB settings after that. I do want to explore subclassing the test runner to apply the same behaviour to the portalocker cache `LOCATION` since that is more structurally sound, but for now I went with this bandaid fix, since the django test runner code is quite complicated and I don't know how easy it is to add in this behaviour.

Reply:
@Viicos I tried the above, but I couldn't get it to work reliably and I have to touch/monkepatch a lot of private API of django-redis which I'm not feeling comfortably with.

Additionally, this implicitly makes the redis DB 0 not available, and if DB 1 would be used for celery for example, the test process can only start using DB 2 or higher, which in turn again limits the available number of parallel processes since default redis clusters have 16 DBs. In my case with 16 CPUs on my system, I can't rely on automatic detection for the parallel count because of these arbitrary constraints. You can also configure redis to have more or less databases, so hardcoding these numbers is also annoying.

I'll stick with this approach of explicitly skipping portalocker + assigning an explicit DB index for the portalocker cache, this at least makes things as small as possible and keeps them predictable.

Reply:
I think if tests passes currently with the solution it should probably be fine

Reply:
yeah I'm seeing failures on the flaky hypothesis test now, but in general this patch should fix a bunch of frustrations